src/Pages/LoginPage/Components/LoginForm.jsx

```diff
@@ -0,0 +1,140 @@
+import { useState } from "react";
+import { Link, useNavigate, useLocation } from "react-router-dom";
+import { useModal, useAuth, useAlert } from "../../../CustomHooks/";
+import axios from "axios";
+
+export const LoginForm = (props) => {
+  let navigate = useNavigate();
+  const { authDispatch } = useAuth();
+  const { setShowModal } = useModal();
+  const { pathname } = useLocation();
+  const { setShowAlert } = useAlert();
+  const [error, setError] = useState("");
+  const [showPassword, setShowPassword] = useState(false);
+  const [loginDetails, setLoginDetails] = useState({
+    email: "",
+    password: "",
+  });
+
+  const loginInputHandler = (e) => {
+    setLoginDetails({
+      ...loginDetails,
+      [e.target.name]: e.target.value,
+    });
+  };
+
+  const formSubmitHandler = async (e) => {
+    e.preventDefault();
+    const { email, password } = loginDetails;
+    setError("");
+    try {
+      const response = await axios.post("/api/auth/login", {
+        email,
+        password,
+      });
+      if (response.status === 200) {
+        setError(null);
+        authDispatch({ type: "LOGIN", payload: response.data });
+
+        //Save user info in local storage
+        localStorage.setItem("user", JSON.stringify(response.data));
+
+        //reset on submission
+        setLoginDetails({
+          email: "",
+          password: "",
+        });
+        if (pathname === "/products") {
+          setShowModal(false);
+        }
+        setShowAlert({
+          showAlert: true,
+          alertMessage: "Logged in successfully!",
+          type: "success",
+        });
+        if (pathname === "/login") {
+          navigate(-1);
+          //         setShowModal(false);
+        }
+      }
+    } catch (err) {
+      console.log(err);
+      if (err.response.status === 401) {
+        setError("Invalid email or password. Please try again.");
+      } else if (err.response.status === 404) {
+        setError("No user found with this email. Please try again.");
+      } else {
+        setError("Unexpected error. Please try again in some time.");
+      }
+    }
+  };
+
+  const testLogin = () => {
+    setLoginDetails({
+      ...loginDetails,
+      email: "adarshbalika@gmail.com",
+      password: "adarshBalika123",
+    });
+  };
+
+  const loginClasses = "login__container " + props.classes;
+
+  return (
+    <main className={loginClasses}>
+      <form
+        className="input-form"
+        autoComplete="on"
+        onSubmit={formSubmitHandler}
+      >
+        <h3>Login</h3>
+        <label htmlFor="email">
```

Comment:
have this label wrap the input, that is the common way of providing labels to the input tags.